tmt/utils.py

```diff
@@ -6263,3 +6275,22 @@ def retry(
             logger.fail(str(exc))
             time.sleep(interval)
     raise RetryError(label, causes=exceptions)
+
+
+def get_file_content_from_url(url: str) -> str:
+    try:
+        response = urllib.request.urlopen(url)
+    except urllib.error.URLError as error:
+        raise GeneralError(f'Could not open url {url}') from error
+    content = response.read().decode('utf-8')
+    if not isinstance(content, str):
+        raise GeneralError(f'Could not decode content from url {url}')
+    return content
```

Comment:
A couple of points:

* I'd probably switch to `requests` rather than using `urllib` directly. It would work, but `requests` are fairly common and used in tmt code already.
* we have `retry_session` ready to help with retries and short outages.
* `get_url_content` would sound a bit better to me, especially the "file" seems superfluous to me.

Something like this, maybe?

```python
def get_url_content(url: str) -> str:
    try:
        with retry_session() as session:
            response = session.get(url)

            if response.ok:
                return response.text

    except Exception as error:
        raise GeneralError(f'Could not open url {url}') from error
```

* And @psss will definitely point out missing single quotes & dot in the exception message, so let me do it right away: `f"Could not open url '{url}'."` :)

Reply:
Thank you for the suggestion. Added in 78d7217caecb8c3bd05d92445264de9de4bfaf02.